### ISSUE TITLE
Need reference for PCMAX (default setting for PC in CL apparently)

### ISSUE THREAD

**Issue by jdramsey** [cmu-phil/causal-learn#48]
After some sleuthing in the Python code and some performance testing with the help of Pablo Puig and Bryan Andrews, I've come to the conclusion that the _default_ setting for PC in CL is maxP. That is an algorithm of mine called PCMAX. The only reference for for this algorithm currently in the literature that I know of is an arXiv tech report I put out in 2016, this one:

Ramsey, J. (2016). Improving accuracy and scalability of the pc algorithm by maximizing p-value. arXiv preprint arXiv:1610.00378.

Anyway, the performance of PC in CL is the same as the performance of PCMAX in Tetrad (and different from the performance of PC in Tetrad.)

If this is going to be made the _default_ setting for PC for CL, perhaps this tech report should be referenced for the PC algorithm in the documentation? Otherwise no one will know what the algroithm is. I'll see if I can't sent that to a conference somewhere to get it published. I didn't realize that was being made the default here.

**Comment by jdramsey** [comment]
Sorry, I should say, if you can find a published reference for it, use that. Or at least describe the procedure in the documentation. I think.

**Comment by kunwuz** [comment]
Sure, thanks so much for your suggestions. I've added your paper as the reference for maxP and definiteMaxP (uc_rule). Please let me know if anything is wrong.

BTW, I'm not totally sure that our default setting is PCMAX, since the default value for 'uc_rule' is 0, as follows:

`def pc(
    data: ndarray, 
    alpha=0.05, 
    indep_test=fisherz, 
    stable: bool = True, 
    uc_rule: int = 0, 
    uc_priority: int = 2,
    mvpc: bool = False, 
    correction_name: str = 'MV_Crtn_Fisher_Z',
    background_knowledge: BackgroundKnowledge | None = None, 
    verbose: bool = False, 
    show_progress: bool = True,
    node_names: List[str] | None = None, 
):`

where

> uc_rule: how unshielded colliders are oriented. Default: 0.
       0: run uc_sepset.
       1: run maxP. Orient an unshielded triple X-Y-Z as a collider with an additional CI test.
       2: run definiteMaxP. Orient only the definite colliders in the skeleton and keep track of all the definite non-colliders as well.

It's very likely that I misunderstood something, as I'm not familiar with PCMAX. It might be the case that we are using PCMAX somewhere else?

**Comment by jdramsey** [comment]
Sorry, this is for a problem with 20 nodes, 40 edges, 1000 samples, averaged over 10 runs.

    p = 20
    a = 4
    d = a / (p - 1)
    n = 1000
    runnum = 10

**Comment by jdramsey** [comment]
Sorry, linear, Gaussian, using Fisher Z for PC_CL and PCMAX_CC--but perhaps the tests are behaving differently in the two platforms--I'll look into it...

**Comment by jdramsey** [comment]
Anyway, I am very curious about this. If you are getting PCMAX performance from the PC sepsets step alone (without maxp) that is something I don't know how to do. Wayne Lam did a study of this for his MA thesis in Python and didn't see how to do it either, with several variants of the procedure. So I need to figure out exactly what the CL code is doing. I'll forward future comments offline and close this.

**Comment by jdramsey** [comment]
Yeah I get the same performance whether I set ucrule to 0 or 1. Setting it to 2 gives a better result, close to GRASP_CC but with lower recalls.

`	Alg   AP      AR     AHP   AHPCE     AHR   AHRCE   McAdj   McArr   F1Adj   F1Arr     SHD    E
	1	1.00	0.71	0.95	0.95	0.52	0.73	0.81	0.68	0.83	0.67	12.40	
`
